dydx/dydxViews/dydxViews/_v4/Vault/Landing/dydxVaultChartViewModel.swift

```diff
@@ -0,0 +1,211 @@
+        }
+    }
+    
+    public enum ValueTimeOption: CaseIterable, RadioButtonContentDisplayable {
+        case oneDay
+        case sevenDays
+        case thirtyDays
+        
+        var displayText: String {
+            let path: String
+            switch self {
+            case .oneDay:
+                path = "APP.GENERAL.TIME_STRINGS.1D"
+            case .sevenDays:
+                path = "APP.GENERAL.TIME_STRINGS.7D"
+            case .thirtyDays:
+                path = "APP.GENERAL.TIME_STRINGS._30D"
+            }
+            return DataLocalizer.shared?.localize(path: path, params: nil) ?? ""
+        }
+        
+        fileprivate var valueFormatter: TimeAxisValueFormatter {
+            let formatter = TimeAxisValueFormatter()
+            switch self {
+            case .oneDay:
+                formatter.dateFormat = "HH:mm"
+            case .sevenDays, .thirtyDays:
+                formatter.dateFormat = "HH:mm\nMM/dd"
+            }
+            return formatter
+        }
+    }
+
+    public override func createView(parentStyle: ThemeStyle = ThemeStyle.defaultStyle, styleKey: String? = nil) -> PlatformView {
+        PlatformView(viewModel: self, parentStyle: parentStyle, styleKey: styleKey) { [weak self] _  in
+            guard let self = self else { return AnyView(PlatformView.nilView) }
+            return AnyView(dydxVaultChartView(viewModel: self)).wrappedInAnyView()
+        }
+    }
+}
+
+private struct dydxVaultChartView: View {
+    @ObservedObject var viewModel: dydxVaultChartViewModel
+    
+    private var radioButtonsRow: some View {
+        HStack(spacing: 0) {
+            RadioButtonGroup(selected: $viewModel.selectedValueType,
+                             options: viewModel.valueTypeOptions,
+                             buttonClipStyle: .capsule,
+                             itemWidth: nil,
+                             itemHeight: 40
+            )
+            Spacer()
+            RadioButtonGroup(selected: $viewModel.selectedValueTime,
+                             options: viewModel.valueTimeOptions,
+                             buttonClipStyle: .circle,
+                             itemWidth: 40,
+                             itemHeight: 40
+            )
+        }
+        .padding(.horizontal, 12)
+    }
+    
+    private var chart: some View {
+        viewModel.lineChart.swiftUIView
+    }
+    
+    var body: some View {
+        VStack(spacing: 12) {
+            radioButtonsRow
+            chart
+        }
+    }
+}
+
+// DateTimeAxisFormatter is broken for ONEHOUR and seem to overcomplicate time value formatting so writing custom IAxisValueFormatter
+fileprivate class TimeAxisValueFormatter: DateFormatter, IAxisValueFormatter {
+    func stringForValue(_ value: Double, axis: AxisBase?) -> String {
+        let date = Date(timeIntervalSince1970: value)
+        return self.string(from: date)
+    }
+}
```

Comment:
The alternative `DateTimeAxisFormatter` seems to be doing a lot of work and it did not really match the vault designs for xAxis value formatting so I opted for a custom formatter which uses an out-of-the-box DateFormatter more or less. Unsure why `DateTimeAxisFormatter` is so complicated.